docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,186 @@
+- *POT file (.pot)* - A file containing source messages (English sentences) that will be translated,
+- *translation asset* - a JS file or part of the file with generated translations for one language.
+
+## Writing localizable UI
+
+All *messages* needing localization should be passed to the special CKEditor 5's `t()` function. This function can be retrieved from the editor's `locale` instance: `const { t } = editor.locale;` or from any view method `const t = this.t;` Note that this function should not be used as the `Locale` class method because during the translation process a static code analyzer catches *localizable messages* only from function forms.
+
+The `t()` function as the first argument accepts either a string literal, which will be at the same time the *message ID* and the message string or an object literal containing `id`, `string`, and optional `plural` property. This function can't be called on variables or other expressions for the same reason as above.
+
+As the second argument, the translation function accepts a value or an array of values. These "values" will be used to fill the placeholders in the more advanced translation scenarios. And if the `plural` property is specified then the first value will be used as quantity determining the plural form.
+
+When using the `t()` function, you can create your own *localizable messages* or reuse *messages* created in CKEditor 5 packages your project depends on. In case of reusing *messages*, you won't have to worry about translating them as all work will be done by the CKEditor 5 team and Transifex translators anyway (but your help in translating still will be appreciated).
+
+For simple *localizable messages* use the string form for simplicity:
+
+```js
+t( 'insert emoji' );
+t( 'insert %0 emoji', emojiName );
+t( 'insert %0 emoji', [ emojiName ] );
+```
+
+For more advanced scenarios use plain objects forms:
+
+```js
+t( { string: '%0 emoji', id: 'INSERT_EMOJI' }, 'insert' );
+t( { string: '%0 emoji', plural: '%0 emojis', id: 'EMOJI' }, quantity );
+t( { string: '%1 %0 emoji', plural: '%1 %0 emojis', id: 'INSERT_EMOJIS' }, [ quantity, 'Insert' ] );
+```
+
+### Examples
+
+#### Localizing UI of plugins:
+
+```js
+// ...
+editor.ui.componentFactory.add( 'smilingFaceEmoji', locale => {
+	const command = editor.commands.get( 'insertSmilingFaceEmoji' );
+	const buttonView = new ButtonView( locale );
+
+	// The translation function.
+	const { t } = editor.locale;
+
+  // The localized label.
+	const label = t( 'insert smiling face emoji' );
+
+	buttonView.set( {
+		label,
+		icon: emojiIcon,
+		tooltip: true
+	} );
+
+	buttonView.on( 'execute', () => {
+		editor.execute( 'insertSmilingFaceEmoji' );
+		editor.editing.view.focus();
+	} );
+} );
+// ...
+```
+
+Note that this sample lacks a few parts. To check how to create a complete plugin, for example, check the {@link framework/guides/creating-simple-plugin Creating a simple plugin guide}.
```

Comment:
Love this reference to "Creating a simple plugin" guide :heart:

![<3](https://media.giphy.com/media/fnvxr5iIIPCSeTS2Fz/giphy.gif)

Reply:
I just saw this strange wording :D I'll change it.

Reply:
Maybe: "Check \[how to create a complete CKEditor 5 plugin\](link)."?